test/fail_compilation/retscope2.d

```diff
@@ -162,3 +162,46 @@ void foo800()
     scope Exception e;
     throw e;
 }
+
+/*************************************************/
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/*************************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope2.d(1005): Error: scope variable `p` assigned to `this` with longer lifetime
+---
+*/
+
+#line 1000
+
+class C17428
```

Comment:
This compiles without complaint before this PR (which seems correct):
```
class C17428
{
    int* _p;
}

void main() @safe
{
	int x;
	int* p = &x;
	scope C17428 c;
	c._p = p;
}
```
but does no longer afterwards.